packages/cardano-services/src/Asset/types.ts

```diff
@@ -16,3 +16,15 @@ export interface NftMetadataService {
    */
   getNftMetadata(asset: AssetPolicyIdAndName): Promise<Asset.NftMetadata | undefined>;
 }
+
+/**
+ * Service to get CIP-38? token metadata for a given subject
+ */
+export interface TokenMetadataService {
+  /**
+   * Get CIP-38? token metadata for a given subject
+   *
+   * @returns CIP-38? token metadata, `null` if not found
+   */
+  getTokenMetadata(assetIds: Cardano.AssetId[]): Promise<(Asset.TokenMetadata | null)[]>;
```

Comment:
Since AssetProvider only has getAsset (single asset), this could be simplified.
```suggestion
  getTokenMetadata(assetId: Cardano.AssetId): Promise<Asset.TokenMetadata | null>;
```

Reply:
I think it's better to make the interface match the service capabilities, as we'll be making batch requests if moving from a lazy cache to one based on discovery at the point of minting.